### PR TITLE
refactor: Clean up ignore files

### PR DESCRIPTION
**Pull request by mgax**
Merge all `.gitignore` files in the repository and update `.dockerignore` to match.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,17 +1,16 @@
-*.pyc
-.DS_Store
-*.swp
+# Be sure to update .dockerignore to match .gitignore
 /venv/
 /static/
 /media/
-/docs/_build/
-/ietf/static/node_modules
-/ietf/static/yarn-error.log
```
Comment (kesara) [inline .gitignore:9]: This entry is not in the new updated ignore files.

File: .gitignore [hunk 2]
```diff
@@ -1,17 +1,16 @@
-*.pyc
-.DS_Store
-*.swp
+# Be sure to update .dockerignore to match .gitignore
 /venv/
 /static/
 /media/
-/docs/_build/
```
Comment (kesara) [inline .gitignore:7]: This entry is not in the new updated ignore files, was that deliberate?

**Review by kesara — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (mgax) [reply .gitignore:7]: I couldn't find any `docs` directory in the codebase, either on `main` or in history, so I assumed the rule is not useful.

File: .gitignore (hunk 1 above)
Comment (mgax) [reply .gitignore:9]: There's already a generic `*.log` entry that will ignore `/ietf/static/yarn-error.log` too.

**Pull request merged**